### DISCUSSION TITLE
Importing using the import folder suddenly failing

### DISCUSSION THREAD

**Discussion by StianOby** [photoprism/photoprism#2377] [Q&A] [answered]
For months I have been using the import folder (with WebDAV) to import new images to our PhotoPrism album. Recently something has begun failing in the import process, and I can't find any similar errors reported by other user by searching.

The photos I upload to the import folder appear to be correctly renamed and moved to the relevant originals folder. The import folder is also emptied, as expected. However, the imported images are not indexed. Thus, I have to manually index them by reindexing the relevant month folder in Originals.

This happens whether I am using WebDAV through my Ubuntu 22.04 computer (w/Nautilus) or by using the PhotoSync app on my phone. 

_Here is an example log from a recent import, which illustrates the rather cryptic error message I consistently get ("is empty in"):_

 > 2022-05-29 19:58:50 INFO import fullført på 14 s
> 
> 2022-05-29 19:58:45 ERRO import: 2022/05/20220514_202609_022547A2.jpg is empty in 2022/05/20220514_202609_022547A2.jpg
> 
> 2022-05-29 19:58:44 INFO import: moving main jpg file IMG_2832.JPG to 2022/05/20220514_202609_022547A2.jpg
> 
> 2022-05-29 19:58:44 INFO media: IMG_2832.JPG was taken at 2022-05-14 20:26:09 +0000 UTC (meta)
> 
> 2022-05-29 19:58:44 ERRO import: 2022/05/20220514_202604_9692FB7E.jpg is empty in 2022/05/20220514_202604_9692FB7E.jpg
> 
> 2022-05-29 19:58:44 INFO import: moving main jpg file IMG_2831.JPG to 2022/05/20220514_202604_9692FB7E.jpg
> 
> 2022-05-29 19:58:44 INFO media: IMG_2831.JPG was taken at 2022-05-14 20:26:04 +0000 UTC (meta)
> 
> 2022-05-29 19:58:44 ERRO import: 2022/05/20220514_202558_02FA5433.jpg is empty in 2022/05/20220514_202558_02FA5433.jpg
> 
> 2022-05-29 19:58:43 INFO import: moving main jpg file IMG_2830.JPG to 2022/05/20220514_202558_02FA5433.jpg
> 
> 2022-05-29 19:58:43 INFO media: IMG_2830.JPG was taken at 2022-05-14 20:25:58 +0000 UTC (meta)
> 
> 2022-05-29 19:58:43 ERRO import: 2022/05/20220514_202529_48BE3D8D.jpg is empty in 2022/05/20220514_202529_48BE3D8D.jpg
> 
> 2022-05-29 19:58:43 INFO import: moving main jpg file IMG_2829.JPG to 2022/05/20220514_202529_48BE3D8D.jpg
> 
> 2022-05-29 19:58:43 INFO media: IMG_2829.JPG was taken at 2022-05-14 20:25:29 +0000 UTC (meta)
> 
> 2022-05-29 19:58:42 ERRO import: 2022/05/20220514_202526_3B61C0FA.jpg is empty in 2022/05/20220514_202526_3B61C0FA.jpg
> 
> 2022-05-29 19:58:42 INFO import: moving main jpg file IMG_2828.JPG to 2022/05/20220514_202526_3B61C0FA.jpg
> 
> 2022-05-29 19:58:42 INFO media: IMG_2828.JPG was taken at 2022-05-14 20:25:26 +0000 UTC (meta)
> 
> 2022-05-29 19:58:42 ERRO import: 2022/05/20220514_202512_0F96CC74.jpg is empty in 2022/05/20220514_202512_0F96CC74.jpg
> 
> 2022-05-29 19:58:42 INFO import: moving main jpg file IMG_2827.JPG to 2022/05/20220514_202512_0F96CC74.jpg
> 
> 2022-05-29 19:58:42 INFO media: IMG_2827.JPG was taken at 2022-05-14 20:25:12 +0000 UTC (meta)
> 
> 2022-05-29 19:58:41 INFO import: found no .ppignore file
> 
> 2022-05-29 19:58:41 INFO classify: loading labels from labels.txt
> 
> 2022-05-29 19:58:36 INFO classify: loading nasnet
> 
> 2022-05-29 19:58:36 INFO kopierer filer fra import
> 
> 2022-05-29 19:53:23 INFO webdav: PUT /import/IMG_2832.JPG
> 
> 2022-05-29 19:53:23 INFO webdav: PUT /import/IMG_2831.JPG
> 
> 2022-05-29 19:53:23 INFO webdav: PUT /import/IMG_2830.JPG
> 
> 2022-05-29 19:53:22 INFO webdav: PUT /import/IMG_2829.JPG
> 
> 2022-05-29 19:53:22 INFO webdav: PUT /import/IMG_2828.JPG
> 
> 2022-05-29 19:53:22 INFO webdav: PUT /import/IMG_2827.JPG

_As you can see below, these files are successfully indexed if I trigger the process manually:_

> 2022-05-29 20:32:33 INFO indeksering fullført på 65 s
> 
> 2022-05-29 20:32:27 INFO index: found no orphan files [91.26861ms]
> 
> 2022-05-29 20:32:26 INFO faces: updated 7 markers, recognized 7 faces, 0 unknown [475.701899ms]
> 
> 2022-05-29 20:32:26 INFO faces: found 7 unmatched markers
> 
> 2022-05-29 20:32:25 INFO index: added main jpg file 2022/05/20220514_202609_022547A2.jpg
> 
> 2022-05-29 20:32:23 INFO media: 20220514_202609_022547A2.jpg was taken at 2022-05-14 20:26:09 +0000 UTC (meta)
> 
> 2022-05-29 20:32:22 INFO media: created 11 thumbnails for 20220514_202609_022547A2 [4.204420521s]
> 
> 2022-05-29 20:32:18 INFO index: added main jpg file 2022/05/20220514_202604_9692FB7E.jpg
> 
> 2022-05-29 20:32:18 INFO index: matched 1 label with 20220514_202604_9692FB7E.jpg [2.192161152s]
> 
> 2022-05-29 20:32:16 INFO media: 20220514_202604_9692FB7E.jpg was taken at 2022-05-14 20:26:04 +0000 UTC (meta)
> 
> 2022-05-29 20:32:15 INFO media: created 11 thumbnails for 20220514_202604_9692FB7E [4.270939832s]
> 
> 2022-05-29 20:32:11 INFO index: added main jpg file 2022/05/20220514_202558_02FA5433.jpg
> 
> 2022-05-29 20:32:11 INFO index: matched 1 label with 20220514_202558_02FA5433.jpg [2.058082393s]
> 
> 2022-05-29 20:32:09 INFO media: 20220514_202558_02FA5433.jpg was taken at 2022-05-14 20:25:58 +0000 UTC (meta)
> 
> 2022-05-29 20:32:09 INFO index: found 2 faces in 20220514_202558_02FA5433.jpg [3.148446071s]
> 
> 2022-05-29 20:32:05 INFO media: created 11 thumbnails for 20220514_202558_02FA5433 [4.1950762s]
> 
> 2022-05-29 20:32:01 INFO index: added main jpg file 2022/05/20220514_202529_48BE3D8D.jpg
> 
> 2022-05-29 20:32:01 INFO index: matched 1 label with 20220514_202529_48BE3D8D.jpg [2.08749931s]
> 
> 2022-05-29 20:31:59 INFO media: 20220514_202529_48BE3D8D.jpg was taken at 2022-05-14 20:25:29 +0000 UTC (meta)
> 
> 2022-05-29 20:31:59 INFO index: found 2 faces in 20220514_202529_48BE3D8D.jpg [3.16483644s]
> 
> 2022-05-29 20:31:56 INFO media: created 11 thumbnails for 20220514_202529_48BE3D8D [4.146098811s]
> 
> 2022-05-29 20:31:51 INFO index: added main jpg file 2022/05/20220514_202526_3B61C0FA.jpg
> 
> 2022-05-29 20:31:51 INFO index: matched 3 labels with 20220514_202526_3B61C0FA.jpg [2.096187882s]
> 
> 2022-05-29 20:31:49 INFO media: 20220514_202526_3B61C0FA.jpg was taken at 2022-05-14 20:25:26 +0000 UTC (meta)
> 
> 2022-05-29 20:31:49 INFO index: found 2 faces in 20220514_202526_3B61C0FA.jpg [3.186640017s]
> 
> 2022-05-29 20:31:46 INFO media: created 11 thumbnails for 20220514_202526_3B61C0FA [3.753869586s]
> 
> 2022-05-29 20:31:42 INFO index: added main jpg file 2022/05/20220514_202512_0F96CC74.jpg
> 
> 2022-05-29 20:31:42 INFO index: matched 1 label with 20220514_202512_0F96CC74.jpg [5.944082494s]
> 
> 2022-05-29 20:31:36 INFO media: 20220514_202512_0F96CC74.jpg was taken at 2022-05-14 20:25:12 +0000 UTC (meta)
> 
> 2022-05-29 20:31:36 INFO index: found 1 face in 20220514_202512_0F96CC74.jpg [3.332634918s]
> 
> 2022-05-29 20:31:33 INFO faces: loading facenet
> 
> 2022-05-29 20:31:33 INFO media: created 11 thumbnails for 20220514_202512_0F96CC74 [4.162800336s]
> 
> 2022-05-29 20:31:28 INFO index: found no .ppignore file
> 
> 2022-05-29 20:31:28 INFO indekserer filer i 2022/05

**Has anyone experienced this error and know how to fix it? Or have suggestions on how to troubleshoot?**

**Comment by lastzero** [comment, marked as answer]
It seems that the error you see in the logs is generated here:

https://github.com/photoprism/photoprism/blob/e1f14b7ec2076c4ec6bcd4bc52b54d3504c932e2/internal/photoprism/import_worker.go#L124

Of course, that should never happen and we don't observe it here. So I suspect it's a timing and/or concurrency issue, or there have been changes in the underlying Go and/or OS libraries.

**Would be great if you could answer these questions so we can get a better picture of your specific environment:**

1. Is the problem occurring with the latest stable Docker image or a customized/older build?
2. What file systems are you using?
3. Are you using virtualization technologies such as virtual machines or networking/encrypted file systems?
4. Is the entire environment running on a single Linux machine or is it a more complex tech stack, e.g., a mix of multiple operating systems and servers, such as a NAS providing networked storage?

It will also be helpful to know if it makes a difference if you **(a)** copy the imported files instead of moving them (disable "Move Files" in the UI under *Library* > *Import*), and **(b)** reduce the number of workers to 1 in the `docker-compose.yml` config (restart for changes to take effect).

- If the server uses network storage or a virtual file system, there could theoretically be a small delay before the size of the moved files is visible/updated. A solution could then be to wait a short time and try again, or to change the file system settings in Linux or Docker so that  files and their metadata are always synchronized.
- Note related files are imported together. So if for some reason the same sidecar file is grouped with two different other files, you may also get strange errors. Your logs don't look like that's the case, though.

**Reply by StianOby** [reply]
Thanks for digging into this – and sorry for the late reply. I have been away from home for about a week, so only had time to reply to your questions now.

**1)** I am using the latest stable build, yes. At least I think I am. It is updated automatically using the suggested docker method (Watchtower). The current build number shown in settings is "Build [220528-efb5d710]"

**2)** I am running Photoprism using the default docker image, and it is running on a linux computer that I use as a house server/media center. The local filesystem on that machine is ext4. The import and originals folders, however, are on my Synology NAS. The folder containing the import and originals folders is mounted on the house server through SMB as /media/PhotoPrism/. Here is the relevant line of /etc/fstab (with *** being my redactions):
`//192.168.1.***/PhotoPrism /media/PhotoPrism cifs credentials=/home/***/.smbcredentials,uid=1000,gid=1000 0 0`

**3)** No, at least not as far as I am aware. The OS on the machine running PhotoPrism is Ubuntu LTS (20.04.4). I have not set up any virtualization or encryption on my own, and believe I actively avoided it on the initial install.

**4)** As mentioned above, the import and originals folders reside on a Synology NAS, and are mounted using SMB.


**Unchecking "Move Files" makes no difference:**
```
2022-06-08 21:08:06 INFO import fullført på 5 s

2022-06-08 21:08:01 ERRO import: 2022/06/20220602_102542_280178B6.jpg is empty in 2022/06/20220602_102542_280178B6.jpg

2022-06-08 21:08:01 INFO import: moving main jpg file 20220602_122542.jpg to 2022/06/20220602_102542_280178B6.jpg

2022-06-08 21:08:01 INFO media: 20220602_122542.jpg was taken at 2022-06-02 10:25:42 +0000 UTC (meta)

2022-06-08 21:08:00 INFO import: found no .ppignore file

2022-06-08 21:08:00 INFO kopierer filer fra import

2022-06-08 21:07:53 INFO webdav: PUT /import/20220602_122542.jpg
```

**Setting the number of workers to 1 also makes no difference (I did restart both the docker container and the entire home server):**
```
2022-06-08 21:17:54 INFO import fullført på 11 s

2022-06-08 21:17:48 ERRO import: 2022/06/20220604_120636_7B80BA87.jpg is empty in 2022/06/20220604_120636_7B80BA87.jpg

2022-06-08 21:17:48 INFO import: moving main jpg file 20220604_140635.jpg to 2022/06/20220604_120636_7B80BA87.jpg

2022-06-08 21:17:48 INFO media: 20220604_140635.jpg was taken at 2022-06-04 12:06:36 +0000 UTC (meta)

2022-06-08 21:17:48 INFO import: 2022/06/20220602_102542_280178B6.jpg already exists

2022-06-08 21:17:48 INFO media: 20220602_122542.jpg was taken at 2022-06-02 10:25:42 +0000 UTC (meta)

2022-06-08 21:17:48 INFO import: found no .ppignore file

2022-06-08 21:17:48 INFO classify: loading labels from labels.txt

2022-06-08 21:17:43 INFO classify: loading nasnet

2022-06-08 21:17:43 INFO kopierer filer fra import

2022-06-08 21:17:35 INFO webdav: PUT /import/20220604_140635.jpg
```

**Reply by jb8628** [reply]
I'm experiencing the same errors with an originals folder provided via SMB from a TrueNAS system. My docker host is Debian 11. The error does not occur if I mount an originals folder via NFS instead. Has anyone found a way to make this work with SMB?

**Reply by lastzero** [reply]
I suspect this needs to be changed/fixed on the SMB server rather than on the client side or in PhotoPrism.

**Reply by jb8628** [reply]
Thanks! That was my thought as well. I've played around with various SMB parameters without success (can't say I'm an expert on this). Interesting though that the issue seems to occur when using both Synology (@StianOby ) and TrueNAS (me).

**Reply by StianOby** [reply]
Well, I could always try using NFS. But did you notice that I have had PhotoPrism, including the import feature, working as expected for many months before this started occurring? And, if I remember correctly, there were no updates to the NAS software in the period this behavior started occurring.

**Reply by lastzero** [reply]
At best it's related to a new Linux base image, that's why we not only provide the standard version based on Debian 12 but also Debian 11 and Ubuntu LTS. Check our repository on Docker Hub.

**Reply by StianOby** [reply]
Can confirm that mounting with NFS instead of using SMB makes the issue go away. This is an acceptable workaround for me, but if someone notices that the issue is fixed sometime down the line, please bump the thread.
